File: src/main/java/name/gyger/jmoney/account/EntryController.java
package name.gyger.jmoney.account;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/rest/accounts")
public class EntryController {

    private final Logger log = LoggerFactory.getLogger(EntryController.class);

    private final EntryService entryService;

    public EntryController(EntryService entryService) {
        this.entryService = entryService;
    }

    @RequestMapping(path = "/{accountId}/entries/count", method = RequestMethod.GET)
    public long getEntryCount(@PathVariable long accountId) {
        return entryService.getEntryCount(accountId);
    }

    @RequestMapping(path = "/{accountId}/entries", method = RequestMethod.GET)
    public List<Entry> getEntries(@PathVariable long accountId,
                                     @RequestParam(value = "page", required = false) Integer page,
                                     @RequestParam(value = "filter", required = false) String filter) {
        return entryService.getEntries(accountId, page, filter);
    }

    @RequestMapping(path = "/{accountId}/entries/{entryId}", method = RequestMethod.GET)
    public Entry getEntry(@PathVariable long accountId, @PathVariable long entryId) {
        return entryService.getEntry(entryId);
    }

    @RequestMapping(path = "/{accountId}/entries", method = RequestMethod.POST)
    public long createEntry(@RequestBody Entry entry, @PathVariable long accountId) {
        entry.setAccountId(accountId);
        return entryService.createEntry(entry);
    }

    @RequestMapping(path = "/{accountId}/entries/{entryId}", method = RequestMethod.PUT)
    public void updateEntry(@PathVariable long accountId, @PathVariable long entryId, @RequestBody Entry entry) {
        entry.setId(entryId);
        entry.setAccountId(accountId);
        entryService.updateEntry(entry);
    }

    @RequestMapping(path = "/{accountId}/entries/{entryId}", method = RequestMethod.DELETE)
    public void deleteEntry(@PathVariable long accountId, @PathVariable long entryId) {
        entryService.deleteEntry(entryId);
    }

}

File: src/main/kotlin/name/gyger/jmoney/account/EntryController.kt
package name.gyger.jmoney.account

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/rest/accounts")
class EntryController(private val entryService: EntryService,
                      private val entryRepository: EntryRepository) {

    @GetMapping("/{accountId}/entries/count")
    fun getEntryCount(@PathVariable accountId: Long): Long {
        return entryRepository.count(accountId)
    }

    @GetMapping("/{accountId}/entries")
    fun getEntries(@PathVariable accountId: Long,
                   @RequestParam(value = "page", required = false) page: Int?,
                   @RequestParam(value = "filter", required = false) filter: String?): List<Entry> {
        return entryService.getEntries(accountId, page, filter)
    }

    @GetMapping("/{accountId}/entries/{entryId}")
    fun getEntry(@PathVariable accountId: Long, @PathVariable entryId: Long): Entry {
        return entryService.getEntry(entryId)
    }

    @PostMapping("/{accountId}/entries")
    fun createEntry(@RequestBody entry: Entry, @PathVariable accountId: Long): Long {
        entry.accountId = accountId
        return entryService.deepSave(entry).id
    }

    @PutMapping("/{accountId}/entries/{entryId}")
    fun updateEntry(@PathVariable accountId: Long, @PathVariable entryId: Long, @RequestBody entry: Entry) {
        entry.id = entryId
        entry.accountId = accountId
        entryService.deepSave(entry)
    }

    @DeleteMapping("/{accountId}/entries/{entryId}")
    fun deleteEntry(@PathVariable accountId: Long, @PathVariable entryId: Long) {
        entryRepository.delete(entryId)
    }

}
